Harvest.Core/Services/ProjectHistoryService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Harvest.Core.Data;
+using Harvest.Core.Domain;
+
+namespace Harvest.Core.Services
+{
+    public interface IProjectHistoryService
+    {
+        Task<ProjectHistory> AddProjectHistory(Project project, string action, string description, object detailsToSerialize, User user = null);
+    }
+
+    public class ProjectHistoryService : IProjectHistoryService
+    {
+        private readonly IUserService _userService;
+        private readonly AppDbContext _dbContext;
+        private readonly JsonSerializerOptions _jsonOptions;
+
+        public ProjectHistoryService(IUserService userService, AppDbContext dbContext, JsonSerializerOptions jsonOptions)
```

Comment:
Ooh this is nice -- i'm guessing that this is injected based on the defaults you pass to the mvc/controllers method AddJsonOptions?

Reply:
Not from `AddJsonOptions`, but `services.AddScoped(provder => StandardJsonOptions.GetOptions());`. I believe AddJsonOptions only configures for payloads being passed to/from controller actions. But the options for all of them come from `StandardJsonOptions.GetOptions`.